File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        output = [0] * len(temperatures)

        for i in range(len(temperatures) - 1, -1, -1):
            temp = temperatures[i]

            if (not stack):
                stack.append((temp, i))
                continue
            
            while (stack and temp >= stack[-1][0]):
                stack.pop()

            if (stack):
                output[i] = stack[-1][1] - i
            
            stack.append((temp, i))

        return output
            


        

File: 739-daily-temperatures/daily-temperatures.js
var dailyTemperatures = function(temperatures) {
    let output = new Array(temperatures.length).fill(0)
    let stack = []

    for (let i = temperatures.length - 1; i >= 0; i--) {
        let current = temperatures[i]
        if (!stack.length) {
            stack.push([current, i])        
        }
        else {
            while (stack.length && current >= stack.at(-1)[0]) {
                stack.pop()
            }
            if (stack.length) {
                output[i] = stack.at(-1)[1] - i
            }
            stack.push([current, i])
        }
    } 

    return output
};